internal/daemon/sfc-reconciler/sfc.go

```diff
@@ -171,6 +207,47 @@ func (r *SfcReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Handle deletion
+	if sfc.DeletionTimestamp != nil {
+		r.log.Info("ServiceFunctionChain is being deleted", "sfc", sfc.Name)
+
+		// Check if this SFC should be cleaned up on this node
+		matches, err := r.matchesNodeSelector(ctx, sfc.Spec.NodeSelector)
+		if err != nil {
+			r.log.Error(err, "Failed to check node selector during deletion")
+			return ctrl.Result{RequeueAfter: time.Minute}, nil
+		}
+
+		if matches {
+			// Clean up pods created by this SFC
+			if err := r.cleanupNetworkFunctionPods(ctx, sfc); err != nil {
+				r.log.Error(err, "Failed to cleanup network function pods")
+				return ctrl.Result{RequeueAfter: time.Minute}, err
+			}
+		}
+
+		// Remove our finalizer
+		if controllerutil.ContainsFinalizer(sfc, SfcFinalizerName) {
+			controllerutil.RemoveFinalizer(sfc, SfcFinalizerName)
+			if err := r.Update(ctx, sfc); err != nil {
```

Comment:
if inside if inside if inside if.

Can we refactor? At least move all of the finalizer stuff into a function

Reply:
I can add a function. The subsequent commit adds the rbac.